src/Simplex/Chat/Controller.hs

```diff
@@ -193,6 +196,28 @@ data ChatController = ChatController
     logFilePath :: Maybe FilePath
   }
 
+data RemoteHostSession = RemoteHostSession
+  { -- | A process that relays the commands to its host
+    handler :: Async (),
+    -- | Path for local resources to be synchronized with host
+    path :: FilePath,
+    -- | Commands to be relayed to a host
+    command :: forall m . ChatMonad m => ByteString -> m ChatResponse,
+    -- | Fetch a file received by a host
+    fetchFile :: forall m . ChatMonad m => FilePath -> m ChatResponse,
+    -- | Store a file on a host to be sent
+    storeFile :: forall m . ChatMonad m => FilePath -> ByteString -> m ChatResponse
+  }
+
+-- | Host-side dual to RemoteHostSession, on-methods represent HTTP API.
+data RemoteControllerSession = RemoteControllerSession
+  { -- | A process that relays commands and data from remote controller
+    handler :: Async (),
+    onCommand :: forall m . ChatMonad m => ByteString -> m ChatResponse,
```

Comment:
same here - why they are not just functions?

Reply:
Lots of libraries are using and it makes it unnecessary hard to reason about what these functions do, as you need to find where this object is created instead of just finding the function by name. We should not use this pattern.